impress-org/givewp
# Refactor: Update welcome banner with NextGen 3.0 release design

## Description
I've updated the welcome banner with the new 3.0 release design. The design consists of two links, one internal that leads to the VFB new form page and the other which is an external that opens up a new window for the 3.0 doc page. Two videos have replaced the images & include pause and play actions. The banner will continue to remain closed when exited and should only display on the plugin page.

I felt it would be helpful and more organized to redo the banner with React, as it was previously rendered through a php file. Although not much javascript is needed in this particular banner, using this approach has made it easier to read, debug and ultimately reuse in the future.

To accomplish some of the changes into the react app, I've created a single endpoint to handle the new meta key option ``` givewp_welcome_banner_dismiss ```  which determines if the banner has previously been dismissed. A new window object specifically for the WelcomeBanner to get the necessary information to perform the ajax requests and handle the SVG icons from the assets folder. The other changes are from migrating php into jsx.

I have also removed the previous 3.0 service provider and related content that was used prior to branching off the core plugin and have included it in our main service provider work.

## Additional Context
- 3.0 Doc page is not live yet.
- Updated videos are currently in progress. We have fallback images for the video players until they are completed.

## Affects

Plugin Page

## Visuals

<img width="1724" alt="Screen Shot 2023-09-01 at 10 20 39 AM" src="https://github.com/impress-org/givewp/assets/75056371/6ae33b5d-6c3c-4cb0-8777-c9d6016141c6">

## Testing Instructions
- Visit Plugin Page
- View Banner
- Play video if available
- Close banner
- Refresh page - verify banner remains closed

## Pre-review Checklist






---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205399260808646
